source/Octopus.Tentacle.Kubernetes.Tests.Integration/KubernetesClusterOneTimeSetUp.cs

```diff
@@ -18,12 +18,12 @@ public async Task OneTimeSetUp()
         await installer.Install();
 
         //if we are not running in TeamCity, then we need to find the latest local tag and use that if it exists 
-        if (!TeamCityDetection.IsRunningInTeamCity())
+        if (!TeamCityDetection.IsRunningInTeamCity() && bool.TryParse(Environment.GetEnvironmentVariable("KubernetesAgentTests_UseLocalImage"), out var useLocal) && useLocal)
         {
             var imageLoader = new DockerImageLoader(KubernetesTestsGlobalContext.Instance.TemporaryDirectory, KubernetesTestsGlobalContext.Instance.Logger, kindExePath);
             KubernetesTestsGlobalContext.Instance.TentacleImageAndTag = imageLoader.LoadMostRecentImageIntoKind(installer.ClusterName);
         }
-        else
+        else if(TeamCityDetection.IsRunningInTeamCity())
```

Comment:
So we don't need an `else` branch?

It might be clearer if we extract a method that returns `TentacleImageAndTag`, and then set it in this method like:

`KubernetesTestsGlobalContext.Instance.TentacleImageAndTag = GetTentacleImageAndTag()`

Reply:
yeah, fair